ivy/functional/frontends/numpy/mathematical_functions/rounding.py

```diff
@@ -0,0 +1,25 @@
+import ivy
+
+
+def ceil(
+    x,
+    /,
+    out=None,
+    *,
+    where=True,
+    casting="same_kind",
+    order="k",
+    dtype=None,
+    subok=True,
+    signature=None,
+    extobj=None
+):
+    if dtype:
+        x = ivy.astype(ivy.array(x), ivy.as_ivy_dtype(dtype))
+    ret = ivy.ceil(x, out=out)
+    if ivy.is_array(where):
+        ret = ivy.where(where, ret, ivy.default(out, ivy.zeros_like(ret)), out=out)
```

Comment:
After checking the document of numpy.ceil, I found that where, as an array of boolean, is broadcast over the input. At locations where the condition is True, the out array(with the same size as input array x) will be set to the ceil result. Elsewhere, the out array will retain its original value. Note that if an uninitialized out array is created via the default out=None, locations within it where the condition is False will remain uninitialized.
So it seems like your current implementation may not do the same thing as the origenal ceil function.

Reply:
Also, if you can , please try and take the subok, extobj and signature arguments into account. Thank you.

Reply:
I think the problem is that numpy when broadcasting the results with "where" they leave the locations where the condition is False uninitialized while what it is done here is that the uninitialized results are filled with 0 instead of NaN